plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/ProductCollectionController.php

```diff
@@ -1945,6 +832,7 @@ protected function register_collection_handlers( $collection_name, $build_query,
 			'editor_args'   => $editor_args,
 			'preview_query' => $preview_query,
 		);
+		$this->query_builder->set_collection_handler_store( $this->collection_handler_store );
```

Comment:
We want to set the `collection_handler_store` in the `ProductQueryBuilder` class too.